### DISCUSSION TITLE
Spreading Sidekiq across machines. How?

### DISCUSSION THREAD

**Discussion by ralfr** [mastodon/mastodon#21176] [Q&A]
I'll soon reach an instance size, where I'd likely have to move Sidekiq to a different machine. Can anybody explain how to do so?

Unfortunately, the documentation on this is super sparse and given the current success of Mastodon, it would be super helpful if anybody could shed some light on this and share the steps required to do so.

**Comment by rvleij** [comment]
I'm not a specialist in any way, but looking at the way the docker (compose) setup is written, sidekiq uses redis and postgres to share state and find it's joins. So the web FE can run somewhere else as long as redis, postgres and the file storage (config filea and such) are shared across those processes I think.

Also, it should be possible to use Docker Swarm to spread out the containers in a cluster, as long as they can share redis and postgres (which themselves can be on a different host).
